test/AxelarGatewayUpgrade.js

```diff
@@ -0,0 +1,180 @@
+} = ethers;
+const { expect } = chai;
+const {
+    isHardhat,
+    waitFor,
+    getAddresses,
+    getRandomID,
+    getChainId,
+    getGasOptions,
+    buildCommandBatch,
+    getApproveContractCall,
+    getWeightedAuthDeployParam,
+    getWeightedProxyDeployParams,
+    getSignedWeightedExecuteInput,
+    getPayloadAndProposalHash,
+} = require('./utils');
+const { getBytecodeHash } = require('@axelar-network/axelar-contract-deployments');
+
+const getWeights = ({ length }, weight = 1) => Array(length).fill(weight);
+
+describe('AxelarGatewayUpgrade', () => {
+    const threshold = isHardhat ? 4 : 2;
+
+    let ownerWallet;
+    let governanceAddress;
+
+    let interchainGovernanceFactory;
+    let interchainGovernance;
+    let gatewayFactory;
+    let authFactory;
+    let gatewayProxyFactory;
+    let tokenDeployerFactory;
+
+    let auth;
+    let tokenDeployer;
+    let gateway;
+
+    let wallets;
+    let owner;
+    let operators;
+    let mintLimiter;
+
+    const governanceChain = 'Governance Chain';
+
+    before(async () => {
+        wallets = await ethers.getSigners();
+        ownerWallet = wallets[0];
+        mintLimiter = ownerWallet;
+        governanceAddress = wallets[1].address;
+        operators = sortBy(wallets.slice(0, threshold), (wallet) => wallet.address.toLowerCase());
+
+        interchainGovernanceFactory = await ethers.getContractFactory('InterchainGovernance', ownerWallet);
+
+        gatewayFactory = await ethers.getContractFactory('AxelarGateway', owner);
+        authFactory = await ethers.getContractFactory('AxelarAuthWeighted', owner);
+        gatewayProxyFactory = await ethers.getContractFactory('AxelarGatewayProxy', owner);
+        tokenDeployerFactory = await ethers.getContractFactory('TokenDeployer', owner);
+        tokenDeployer = await tokenDeployerFactory.deploy();
+        await tokenDeployer.deployTransaction.wait(network.config.confirmations);
+
+        await deployGateway();
+    });
+
+    const deployGateway = async () => {
+        const buffer = isHardhat ? 10 * 60 * 60 : 10;
+
+        const operatorAddresses = getAddresses(operators);
+
+        auth = await authFactory.deploy(getWeightedAuthDeployParam([operatorAddresses], [getWeights(operatorAddresses)], [threshold]));
+        await auth.deployTransaction.wait(network.config.confirmations);
+
+        const gatewayImplementation = await gatewayFactory.deploy(auth.address, tokenDeployer.address);
+        await gatewayImplementation.deployTransaction.wait(network.config.confirmations);
+
+        const params = getWeightedProxyDeployParams(ownerWallet.address, mintLimiter.address, [], [], threshold);
+
+        const proxy = await gatewayProxyFactory.deploy(gatewayImplementation.address, params);
+        await proxy.deployTransaction.wait(network.config.confirmations);
+
+        await auth.transferOwnership(proxy.address).then((tx) => tx.wait(network.config.confirmations));
```

Comment:
```suggestion
        awaitConfirmation(auth.transferOwnership(proxy.address));
```
Maybe define a helper `awaitReceipt(tx)` that takes the tx promise and awaits on it and wait for the confirmations. It can import the network in utils, so no need to pass in the confirmation as a param?